homeassistant/components/netgear/button.py

```diff
@@ -0,0 +1,78 @@
+"""Support for Netgear Button."""
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN, KEY_COORDINATOR, KEY_ROUTER
+from .router import NetgearRouter, NetgearRouterEntity
+
+
+@dataclass
+class NetgearButtonEntityDescription(ButtonEntityDescription):
+    """Class describing Netgear button entities."""
+
+    action: Callable = lambda router: router
```

Comment:
This default doesn't work. I suggest making a mixin dataclass and have the attribute be required instead so we don't need a default value.